src/main/java/com/algorithmia/algorithmHandler/RequestHandler.java

```diff
@@ -0,0 +1,55 @@
+package com.algorithmia.algorithmHandler;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import java.util.Scanner;
+import org.apache.commons.codec.binary.Base64;
+
+
+final class RequestHandler<INPUT>
+{
+
+    private Scanner input;
+    private JsonParser parser;
+
+    RequestHandler(){
+        this.input = new Scanner(System.in);
+        this.parser = new JsonParser();
+    }
+
+    private Request<INPUT> CreateRequest(String contentType, JsonElement data) throws Exception {
+        if (contentType.equals("json")) {
+            if (data.isJsonArray()) {
+                JsonArray array = data.getAsJsonArray();
+                return new Request<>(contentType, array);
+            }else if(data.isJsonPrimitive() || data.isJsonNull()){
+                Object[] inputs = {data};
+                return new Request<>(contentType, inputs);
+            } else {
+                return new Request<>(contentType, data);
+            }
+        } else if (contentType.equals("text")) {
+            return new Request<>(contentType, data.getAsString());
+        } else if (contentType.equals("binary")) {
+            byte[] bytes = Base64.decodeBase64(data.getAsString());
+            return new Request<>(contentType, bytes);
+        } else {
+            throw new Exception("recieved an invalid content_type.");
+        }
+    }
+
+    Request<INPUT> GetNextRequest() throws Exception{
+        if(input.hasNextLine()){
```

Comment:
`Scanner.hasNextLine()` and `Scanner.nextLine()` are blocking methods; you should wrap those in a `CompletableFuture` and run the subsequent operations when those complete processing (this means that the return type should become something like `CompletableFuture<Request<INPUT>>`.

Reply:
I'm trying hard not to deviate too far from how we did it previously, since we haven't gotten this onto Test yet.
https://github.com/algorithmiaio/langpacks/blob/master/java/pipe/src/main/java/com/algorithmia/Pipe.java#L68

Reply:
Will resolve but feel free to re-open if you think we should deviate from previous Pipe I/O methods.